drivers/ethernet/CMakeLists.txt

```diff
@@ -62,3 +62,4 @@ if(CONFIG_ETH_NATIVE_POSIX)
 endif()
 
 add_subdirectory(phy)
+add_subdirectory(dwc_xgmac)
```

Comment:
The current repository contains numerous files from various Ethernet drivers, creating a cluttered appearance. In an effort to improve organization, I've grouped the XGMAC driver files into a separate folder. Please inform me if this deviates from standard practices, and I'll promptly revert the changes.

Reply:
It's okay, I am already planning to do this for another driver I am working on as well because of different ip versions warranting different files etc